File: selfstudy/js/stars_counthigh.js
/*
task 87:
Stars - CountHigh
You need to count how many values are greater than 50 in star luminosities
*/

function count_above_fifty_stars(brightness) {
  return brightness.filter(x => x > 50).length;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: count_above_fifty_stars([10, 20, 30]) },
  { input: [50, 60, 70], expected: count_above_fifty_stars([50, 60, 70]) },
  { input: [1, 2, 3], expected: count_above_fifty_stars([1, 2, 3]) },
  { input: [100, 200], expected: count_above_fifty_stars([100, 200]) },
  { input: [], expected: count_above_fifty_stars([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('count_above_fifty_stars', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(count_above_fifty_stars(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, count_above_fifty_stars(t.input) === t.expected || JSON.stringify(count_above_fifty_stars(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works

File: selfstudy/python/stars_counthigh.py
'''
task 187:
Stars - CountHigh
You need to count how many values are greater than 50 in star luminosities
'''

def count_above_fifty_stars(brightness):
    return sum(1 for x in brightness if x > 50)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], count_above_fifty_stars([10, 20, 30])),
        ([50, 60, 70], count_above_fifty_stars([50, 60, 70])),
        ([1, 2, 3], count_above_fifty_stars([1, 2, 3])),
        ([100, 200], count_above_fifty_stars([100, 200])),
        ([], count_above_fifty_stars([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = count_above_fifty_stars(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# finally works
